hvplot/interactive.py

```diff
@@ -0,0 +1,362 @@
+        transform = type(self._transform)(self._transform, operator.or_, other, reverse=True)
+        return self._clone(transform)
+    def __rpow__(self, other):
+        other = other._transform if isinstance(other, Interactive) else other
+        transform = type(self._transform)(self._transform, operator.pow, other, reverse=True)
+        return self._clone(transform)
+    def __rrshift__(self, other):
+        other = other._transform if isinstance(other, Interactive) else other
+        transform = type(self._transform)(self._transform, operator.rrshift, other)
+        return self._clone(transform)
+    def __rsub__(self, other):
+        other = other._transform if isinstance(other, Interactive) else other
+        transform = type(self._transform)(self._transform, operator.sub, other, reverse=True)
+        return self._clone(transform)
+    def __rtruediv__(self, other):
+        other = other._transform if isinstance(other, Interactive) else other
+        transform = type(self._transform)(self._transform, operator.truediv, other, reverse=True)
+        return self._clone(transform)
+
+    def plot(self, *args, **kwargs):
+        @pn.depends()
+        def get_ax():
+            from matplotlib.backends.backend_agg import FigureCanvas
+            from matplotlib.pyplot import Figure
+            Interactive._fig = fig = Figure()
+            FigureCanvas(fig)
+            return fig.subplots()
+        kwargs['ax'] = get_ax
+        transform = hv.dim(self._transform, 'plot', accessor=True)
+        return self._clone(transform(*args, **kwargs), plot=True)
+
+    def hvplot(self, *args, **kwargs):
+        transform = type(self._transform)(self._transform, 'hvplot', accessor=True)
+        dmap = 'kind' not in kwargs
+        return self._clone(transform(*args, **kwargs), dmap=dmap)
+
+    #----------------------------------------------------------------
+    # Public API
+    #----------------------------------------------------------------
+
+    @property
+    def layout(self):
+        """
+        Returns a layout of the widgets and output arranged according
+        to the center and and widget location specified in the
+        interactive call.
+        """
+        widget_box = self.widgets
+        panel = self.output
+        loc = self._loc
+        if loc in ('left', 'right'):
+            widgets = Column(VSpacer(), widget_box, VSpacer())
+        elif loc in ('top', 'bottom'):
+            widgets = Row(HSpacer(), widget_box, HSpacer())
+        elif loc in ('top_left', 'bottom_left'):
+            widgets = Row(widget_box, HSpacer())
+        elif loc in ('top_right', 'bottom_right'):
+            widgets = Row(HSpacer(), widget_box)
+        elif loc in ('left_top', 'right_top'):
+            widgets = Column(widget_box, VSpacer())
+        elif loc in ('left_bottom', 'right_bottom'):
+            widgets = Column(VSpacer(), widget_box)
```

Comment:
Seems like this layout code ought to be pulled out into its own function to make this method shorter and easier to follow.  Plus this functionality seems similar to `interact`, DynamicMaps, and other cases where widgets are generated and added to a layout, and it would be nice to have a consistent approach for controlling those.